src/features/defi/providers/cosmos/components/CosmosManager/Deposit/CosmosDeposit.tsx

```diff
@@ -0,0 +1,135 @@
+import { Status } from './components/Status'
+import { CosmosDepositActionType } from './DepositCommon'
+import { DepositContext } from './DepositContext'
+import { initialState, reducer } from './DepositReducer'
+
+export const CosmosDeposit = () => {
+  const translate = useTranslate()
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+  const assetNamespace = 'slip44'
+  const assetId = toAssetId({ chainId, assetNamespace, assetReference })
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
+
+  // user info
+  const { state: walletState } = useWallet()
+  const loading = useSelector(selectPortfolioLoading)
+
+  const validatorData = useAppSelector(state => selectValidatorByAddress(state, contractAddress))
+
+  const apr = useMemo(() => bnOrZero(validatorData?.apr).toString(), [validatorData])
+
+  const opportunities = useCosmosSdkStakingBalances({ assetId })
+  const cosmosOpportunity = useMemo(
+    () =>
+      opportunities?.cosmosSdkStakingOpportunities?.find(
+        opportunity => opportunity.address === contractAddress,
+      ),
+    [opportunities, contractAddress],
+  )
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!cosmosOpportunity) return
+
+        const chainAdapterManager = getChainAdapterManager()
+        const chainAdapter = chainAdapterManager.get(chainId)
+        if (!(walletState.wallet && contractAddress && chainAdapter && apr)) return
+
+        const address = await chainAdapter.getAddress({ wallet: walletState.wallet })
+
+        dispatch({ type: CosmosDepositActionType.SET_USER_ADDRESS, payload: address })
+        dispatch({
+          type: CosmosDepositActionType.SET_OPPORTUNITY,
+          payload: { ...cosmosOpportunity },
+        })
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('CosmosDeposit error:', error)
+      }
+    })()
+  }, [chainId, cosmosOpportunity, apr, contractAddress, walletState.wallet])
+
+  const handleBack = () => {
+    history.push({
+      pathname: location.pathname,
+      search: qs.stringify({
+        ...query,
+        modal: DefiAction.Overview,
+      }),
+    })
+  }
+
+  const StepConfig: DefiStepProps = useMemo(() => {
+    return {
+      [DefiStep.Info]: {
+        label: translate('defi.steps.deposit.info.title'),
+        description: translate('defi.steps.deposit.info.description', { asset: asset.symbol }),
+        component: Deposit,
+      },
+      [DefiStep.Confirm]: {
+        label: translate('defi.steps.confirm.title'),
+        component: Confirm,
+      },
+      [DefiStep.Status]: {
+        label: 'Status',
+        component: Status,
+      },
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
removable?

Reply:
Will tackle it on said stacked PR

Reply:
https://github.com/shapeshift/web/pull/2324